docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -72,6 +72,13 @@ space. By default {es} will automatically create a mapping for every field in
 every document it indexes, but you can switch off this behaviour to
 <<explicit-mapping,take control of your mappings>>.
 
+Moreover every segment requires a small amount of heap memory for each mapped
+field. This per-segment-per-field heap overhead includes a copy of the field
+name, encoded using UTF-8. Usually this is not noticeable, but you may need to
+account for this overhead if you have very long field names or high field
```

Comment:
I wonder if it should provide an indication of how many is a _high field count_?

Reply:
I was kind of being deliberately vague here because I don't really want to work out how much of these overheads are covered by the other sizing guidance (particularly the 1kiB-per-field-per-index rule) particularly since the high segment count in this case was caused by a bug, and the duplication of field names in the heap is also a bug, and fixing either of these bugs would mean we can remove this paragraph.

Reply:
Got it, thanks.